src/migrations/dapps/1739898312357_trades-mv.ts

```diff
@@ -0,0 +1,259 @@
+        WHERE LOWER(idx.address) IN (
+            '${marketplacePolygon.address}',
+            '${marketplaceEthereum.address}'
+        )
+    ) AS si_contract
+    ON t.network = si_contract.network
+
+    WHERE t.type IN ('public_item_order', 'public_nft_order')
+    GROUP BY
+        t.id,
+        t.type,
+        t.created_at,
+        t.network,
+        t.chain_id,
+        t.signer,
+        t.checks,
+        si_contract.index,
+        si_signer.index;
+    `
+  )
+  pgm.addIndex(materializedViewName, ['id'], { name: 'idx_mv_trades_id', unique: true, ifNotExists: true })
+  pgm.sql(`
+    CREATE OR REPLACE FUNCTION refresh_trades_mv()
+        RETURNS TRIGGER
+        LANGUAGE plpgsql
+        AS $$
+        BEGIN
+        REFRESH MATERIALIZED VIEW CONCURRENTLY ${materializedViewName};
+        RETURN NULL;
+    END;
+    $$;
+  `)
+
+  pgm.sql(`
+    -------------------------------------------------------------------------------
+    -- marketplace.trades
+    -------------------------------------------------------------------------------
+    CREATE TRIGGER refresh_trades_mv_on_trades
+    AFTER INSERT OR UPDATE OR DELETE
+    ON marketplace.trades
+    FOR EACH STATEMENT
+    EXECUTE FUNCTION refresh_trades_mv();
+
+    -------------------------------------------------------------------------------
+    -- marketplace.trade_assets
+    -------------------------------------------------------------------------------
+    CREATE TRIGGER refresh_trades_mv_on_trade_assets
+    AFTER INSERT OR UPDATE OR DELETE
+    ON marketplace.trade_assets
+    FOR EACH STATEMENT
+    EXECUTE FUNCTION refresh_trades_mv();
+
+    -------------------------------------------------------------------------------
+    -- marketplace.trade_assets_erc721
+    -------------------------------------------------------------------------------
+    CREATE TRIGGER refresh_trades_mv_on_trade_assets_erc721
+    AFTER INSERT OR UPDATE OR DELETE
+    ON marketplace.trade_assets_erc721
+    FOR EACH STATEMENT
+    EXECUTE FUNCTION refresh_trades_mv();
+
+    -------------------------------------------------------------------------------
+    -- marketplace.trade_assets_erc20
+    -------------------------------------------------------------------------------
+    CREATE TRIGGER refresh_trades_mv_on_trade_assets_erc20
+    AFTER INSERT OR UPDATE OR DELETE
+    ON marketplace.trade_assets_erc20
+    FOR EACH STATEMENT
+    EXECUTE FUNCTION refresh_trades_mv();
+
+    -------------------------------------------------------------------------------
+    -- marketplace.trade_assets_item
+    -------------------------------------------------------------------------------
+    CREATE TRIGGER refresh_trades_mv_on_trade_assets_item
+    AFTER INSERT OR UPDATE OR DELETE
+    ON marketplace.trade_assets_item
+    FOR EACH STATEMENT
+    EXECUTE FUNCTION refresh_trades_mv();
+
+    -------------------------------------------------------------------------------
```

Comment:
Should we create triggers for the squid ones? We could add them for testing purposes if you need to use them in the zone environment.